File: pages/login_page.py
"""Page object for the Login screen (/login)."""
from __future__ import annotations

from playwright.sync_api import Page

from pages.base_page import BasePage


class LoginPage(BasePage):
    path = "/login"

    def __init__(self, page: Page):
        super().__init__(page)
        self.email_input = page.locator("input[name='email']")
        self.password_input = page.locator("input[name='password']")
        self.submit_button = page.locator("form button[type='submit']")
        self.google_button = page.locator("#googleSignIn")
        self.forgot_link = page.get_by_role("link", name="Forgot Password?")
        self.signup_link = page.get_by_role("link", name="Create an Account")

    # --- queries -------------------------------------------------------------

    def is_loaded(self) -> bool:
        return (
            self.email_input.is_visible()
            and self.password_input.is_visible()
            and self.submit_button.is_visible()
        )

    # --- actions -------------------------------------------------------------

    def sign_in(self, email: str, password: str) -> None:
        """Fill the credentials and submit (real Firebase JS auth runs client-side)."""
        self.email_input.fill(email)
        self.password_input.fill(password)
        self.submit_button.click()

    def go_to_signup(self) -> None:
        self.signup_link.click()

    def go_to_forgot_password(self) -> None:
        self.forgot_link.click()

File: src/pages/login-page.ts
/** Page object for the Login screen (/login). */
import { Page, Locator } from '@playwright/test';
import { BasePage } from './base-page';

export class LoginPage extends BasePage {
  path = '/login';

  readonly emailInput: Locator;
  readonly passwordInput: Locator;
  readonly submitButton: Locator;
  readonly googleButton: Locator;
  readonly forgotLink: Locator;
  readonly signupLink: Locator;

  constructor(page: Page) {
    super(page);
    this.emailInput = page.locator("input[name='email']");
    this.passwordInput = page.locator("input[name='password']");
    this.submitButton = page.locator("form button[type='submit']");
    this.googleButton = page.locator('#googleSignIn');
    this.forgotLink = page.getByRole('link', { name: 'Forgot Password?' });
    this.signupLink = page.getByRole('link', { name: 'Create an Account' });
  }

  // --- queries ---------------------------------------------------------------

  async isLoaded(): Promise<boolean> {
    return (
      (await this.emailInput.isVisible()) &&
      (await this.passwordInput.isVisible()) &&
      (await this.submitButton.isVisible())
    );
  }

  // --- actions ---------------------------------------------------------------

  /** Fill the credentials and submit (real Firebase JS auth runs client-side). */
  async signIn(email: string, password: string): Promise<void> {
    await this.emailInput.fill(email);
    await this.passwordInput.fill(password);
    await this.submitButton.click();
  }
}
